src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    internal abstract partial class AbstractSyncNamespaceCodeRefactoringProvider<TService, TNamespaceDeclarationSyntax, TCompilationUnitSyntax> :
+        CodeRefactoringProvider
+        where TService : AbstractSyncNamespaceCodeRefactoringProvider<TService, TNamespaceDeclarationSyntax, TCompilationUnitSyntax>
+        where TNamespaceDeclarationSyntax : SyntaxNode
+        where TCompilationUnitSyntax : SyntaxNode 
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var textSpan = context.Span;
+            var cancellationToken = context.CancellationToken;            
+
+            var state = await State.CreateAsync((TService)this, document, textSpan, cancellationToken);
+            if (state == null)
+            {
+                return;
+            }
+
+            context.RegisterRefactorings(SyncNamespaceCodeAction.CreateCodeActions((TService)this, state));
+        }
+
+        private static INamespaceSymbol GetDeclaredNamespaceSymbol(TNamespaceDeclarationSyntax node, SemanticModel semanticModel)
+        {
+            return semanticModel.GetDeclaredSymbol(node) as INamespaceSymbol;
```

Comment:
why 'as'?  do the callers check for null?  if so, this should be called TryGetDeclaredSymbol.